File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left=0
        wind=0
        maxl=float('inf')
        for right in range(len(nums)):
            wind+=nums[right]
            while wind>=target:
                maxl=min(maxl,right-left+1)
                wind-=nums[left]
                left+=1  
        return maxl if maxl!=float('inf') else 0
        

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left=0;
        int wind=0;
        int min=Integer.MAX_VALUE;
        for(int i=0;i<nums.length;i++){
            wind+=nums[i];
            while(wind>=target){
                min=Math.min(min,i-left+1);
                wind-=nums[left];
                left+=1;
                
            }
        }
        if(min==Integer.MAX_VALUE){
            return 0;
        }else{
            return min;
        }
        
    }
}